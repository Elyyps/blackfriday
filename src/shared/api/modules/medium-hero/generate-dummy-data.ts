import { MediumHeroModule } from "./medium-hero";
// Dummy Data
import { headerContent } from "@app/api/core/header-content";
import Oasis from "@assets/oasis.jpg";
import { uniqueSellingPoints } from "@app/api/core/usp";

export const generateMediumHeroDummyData = (): MediumHeroModule => ({
  headerContent: headerContent,
  image: Oasis,
  usps: uniqueSellingPoints,
  bottomMargin: "0px",
  topMargin: "0px",
  id: "1",
  name: "MediumHeroModule"
});
