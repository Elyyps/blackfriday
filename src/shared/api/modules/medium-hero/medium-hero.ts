import { WordPressPostModule } from "../wordpress-module/wordpress-module";
import { IHeaderContentComponentProps } from "@app/core/header-content";
import { UspModule } from "@app/api/core/usp";

export class MediumHeroModule extends WordPressPostModule {
  public headerContent?: IHeaderContentComponentProps;
  public image?: string;
  public usps: UspModule;
}
