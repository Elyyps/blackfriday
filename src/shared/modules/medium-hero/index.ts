export * from "./medium-hero.component";
