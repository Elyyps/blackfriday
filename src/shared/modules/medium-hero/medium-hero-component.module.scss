.medium-hero {
  @media (max-width: $breakpoint-xsmall-max) {
    margin-left: -16px;
    margin-right: -16px;
  }
}

.holder {
  position: relative;
}

.image {
  height: 260px;
  overflow: hidden;
  position: relative;
  border-radius: 2px;
  @media (max-width: $breakpoint-small-max) {
    height: 200px;
  }
  &::after {
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
    content: "";
    background: rgba(0, 0, 0, 0.3);
  }
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
}

.content {
  position: absolute;
  left: 16px;
  right: 16px;
  top: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  @media (max-width: 360px) {
    margin-top: 16px;
    margin-bottom: 16px;
  }
  h1 {
    color: $global-inverse-color;
    text-align: center;
  }
  span:nth-child(n + 3) {
    color: $global-inverse-color;
  }
  h1:last-child {
    margin-bottom: 0;
  }
}

.top {
  padding: 16px 0;
  @media (max-width: $breakpoint-xsmall-max) {
    padding-right: 16px;
    padding-left: 16px;
  }
}
