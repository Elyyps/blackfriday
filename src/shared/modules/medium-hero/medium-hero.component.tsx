import * as React from "react";
import styles from "./medium-hero-component.module.scss";

export interface IMediumHeroComponentProps {
  backButtonText?: string;
  backButtonURL?: string;
  children?: any;
  image?: string;
  title?: string;
}

const MediumHeroComponent = (props: IMediumHeroComponentProps) => (
  <div className={styles["medium-hero"]}>
    <div className="header-single">
      Breadcrumbs komen hier
      <h1>{props.children}</h1>
      <div className="uk-container">
        <div className="header-single__image">{/* <ImageComponent src={props.image} /> */}</div>
      </div>
    </div>
  </div>
);
export { MediumHeroComponent };
