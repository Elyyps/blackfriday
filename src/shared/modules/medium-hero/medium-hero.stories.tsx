import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { MediumHeroComponent } from "./medium-hero.component";

// Dummy Data
import { headerContent } from "@app/api/core/header-content";
import Oasis from "@assets/oasis.jpg";
import { uniqueSellingPoints } from "@app/api/core/usp";

storiesOf("Medium Hero", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <MediumHeroComponent headerContent={headerContent} image={Oasis} usps={uniqueSellingPoints} />
  ));
