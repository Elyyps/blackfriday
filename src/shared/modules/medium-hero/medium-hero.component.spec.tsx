import React from "react";
import { shallow } from "enzyme";
import { MediumHeroComponent } from "./medium-hero.component";

describe("[MediumHero]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<MediumHeroComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
