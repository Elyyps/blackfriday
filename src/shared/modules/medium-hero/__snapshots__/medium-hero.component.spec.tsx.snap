// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MediumHero] should render component without crashing 1`] = `
<div
  className="medium-hero"
>
  <div
    className="header-single"
  >
    Breadcrumbs komen hier
    <h1 />
    <div
      className="uk-container"
    >
      <div
        className="header-single__image"
      />
    </div>
  </div>
</div>
`;
